lib/boutiques_bids_subject_subsetter.rb

```diff
@@ -182,98 +181,113 @@ def setup #:nodoc:
+    folders_name = filenames_wo_ext_by_folder.keys
 
-        file_fullpath_wo_ext = basename.split('.')[0]
+    files_to_exclude_by_folder = Hash.new { |h, k| h[k] = [] }
+    # Iterate over the full subject directory
 
-        next if filenames_wo_ext_by_folder[folder_name].include?(file_fullpath_wo_ext)
-        files_to_exclude_by_folder[folder_name] << file_fullpath
-      end
+    files_in_subject.each do |file_fullpath|
+      dirname, basename = File.split(file_fullpath)
+      # Extract parent_folder name if
+      # for 'sub-123/ses-01/anat/*_T1.json' folder_name is 'anat'
+      folder_name = Pathname.new(dirname).basename.to_s
+      next if !folders_name.include?(folder_name)
 
-      files_to_exclude_by_folder.each_pair do |folder, filenames|
-        self.addlog("File to exclude from the original subject in '#{folder}' folder exclude the following files:")
-        self.addlog("\n- #{filenames.join("\n- ")}")
-      end
+      file_fullpath_wo_ext = basename.split('.')[0]
 
-      return files_to_exclude_by_folder
+      next if filenames_wo_ext_by_folder[folder_name].include?(file_fullpath_wo_ext)
+      files_to_exclude_by_folder[folder_name] << file_fullpath
     end
 
-    # Copy the original subject, allow to not touch the cache when
-    # the extra files will be deleted.
-    def backup_and_copy_subject(subject_name) #:nodoc:
-        bk_name = "#{subject_name}_#{self.id}"
-        File.rename(subject_name, bk_name) if !File.exist?(bk_name)
-        FileUtils.cp_r(bk_name, subject_name) # Attention when 2 times setup solved by rsync `rsync -a -H --delete --chmod see dp_code bk_name/ subject_name`
-        system("rsync","-arH --delete","#{bk_name}/", subject_name)
+    files_to_exclude_by_folder.each_pair do |folder, filenames|
+      self.addlog("File to exclude from the original subject in '#{folder}' folder exclude the following files:")
+      self.addlog("\n- #{filenames.join("\n- ")}")
     end
 
-    # This method removed the extra files from the copied subject.
-    def remove_extra_files(subject_name,files_to_exclude_by_folder) #:nodoc:
-      self.addlog("Remove files from #{subject_name}:\n")
-      removed_files = ""
-      files_to_exclude_by_folder.each_pair do |folder,filenames|
-        folder = "#{subject_name}/#{folder}"
-        filenames.each do |filename|
-          FileUtils.remove_entry(filename) rescue true
-          removed_files += "- #{filename}\n"
-        end
+    return files_to_exclude_by_folder
+  end
+
+  # Copy the original subject, allow to not touch the cache when
+  # the extra files will be deleted.
+  def backup_and_copy_subject(subject_name) #:nodoc:
+      bk_name = "#{subject_name}_#{self.id}"
+      File.rename(subject_name, bk_name) if !File.exist?(bk_name)
+      FileUtils.cp_r(bk_name, subject_name) # Attention when 2 times setup solved by rsync `rsync -a -H --delete --chmod see dp_code bk_name/ subject_name`
+      system("rsync","-arH --delete","#{bk_name}/", subject_name)
```

Comment:
use the other rsync options that I use in `ssh_data_provider.rb` (e.g. chmod etc)